src/pages/index.js

```diff
@@ -1,35 +1,100 @@
 import "./index.css";
 import Card from "../components/Card.js";
 import FormValidator from "../components/FormValidator.js";
-import {
-  initialCards,
-  selectors,
-  formValidationOptions,
-} from "../utils/constants.js";
+import { selectors, formValidationOptions } from "../utils/constants.js";
 import Section from "../components/Section.js";
 import PopupWithForm from "../components/PopupWithForm.js";
 import PopupWithImage from "../components/PopupWithImage.js";
 import UserInfo from "../components/UserInfo.js";
+import PopupConfirm from "../components/PopupWithConfirm.js";
+import Api from "../components/Api.js";
+
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "5782927d-9e6f-49bd-bfea-d56552d67f06",
+    "Content-Type": "application/json",
+  },
+});
+
+function handleDeleteModal(card) {
+  deletePopup.open();
+  deletePopup.setSubmitFunction(() => {
+    api
+      .toDeleteCard(card._id)
+      .then(() => {
+        deletePopup.close();
+        card.removeCard();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+}
+
+function handleLikeButton(card) {
+  if (card.isLiked) {
+    api
+      .likeCard(card._id, card.name, card.link)
+      .then((res) => {
+        console.log(res);
+        card.setLiked(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  } else {
+    api
+      .disLikeCard(card._id)
+      .then((res) => {
+        card.setLiked(res.isLiked);
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  }
+}
+
+const deletePopup = new PopupConfirm("#modal-card-popupdelte");
+deletePopup.setEventListeners();
 
 const profileDescription = document.querySelector(selectors.profileDescription);
 const profileTitle = document.querySelector(selectors.profileTitle);
 
+const cardSection = new Section((data) => {
+  cardSection.addItems(createCard(data));
+}, selectors.cardSection);
+
 // UserInfo
+api.getUserInformation().then((view) => {
+  userInfoData.setUserInfo({
+    name: view.name,
+    description: view.about,
+  });
+  userInfoData.setAvatar(view.avatar);
+});
```

Comment:
You need to `catch` possible errors at the very **end** of any server request.

**Please, correct it everywhere in the code**